cypress/e2e/directory.spec.ts

```diff
@@ -0,0 +1,85 @@
+/// <reference types="cypress" />
+import '../support/commands';
+
+describe('directory infinite scroll', () => {
+  const PAGE_LIMIT = 20;
+  let totalPages = 0;
+
+  before(() => {
+    cy.truncateDatabase();
+
+    cy.createUsers(100);
+
+    cy.request('http://localhost:3000/api/users').then(response => {
+      totalPages = response.body.totalPages;
+    });
+  });
+
+  beforeEach(() => {
+    cy.visit('http://localhost:3000/directory');
+  });
+
+  function checkInfiniteScroll(pages: number, totalPages: number) {
+    for (let i = 1; i <= pages; i++) {
+      cy.get('[data-cy="user-container"]').children().as('ulChildren');
+      i < totalPages ?
+        cy.get('@ulChildren').should('have.length', PAGE_LIMIT * i) :
+        // last page not guaranteed to be full
+        cy.get('@ulChildren').should('have.length.within', PAGE_LIMIT * (i - 1), PAGE_LIMIT * i);
```

Comment:
stylistic choice, but we should reserve ternary statements for values, rather than actual execution.  an if/else would be preferred here